packages/eslint-plugin-kolibri/lib/rules/no-undefined-translator-keys.js

```diff
@@ -0,0 +1,477 @@
+    return null;
+  }
+}
+
+/**
+ * Walks an AST to find all createTranslator exports and declarations
+ * @param {Object} ast - Parsed AST
+ * @returns {Map<string, Set<string>>} Map of export name to Set of message keys
+ */
+function findTranslatorsInAST(ast) {
+  const translators = new Map();
+  const localTranslators = new Map(); // Track non-exported translators for re-export pattern
+
+  function walk(node) {
+    if (!node || typeof node !== 'object') {
+      return;
+    }
+
+    // Pattern 1: export const name = createTranslator(...)
+    if (
+      node.type === 'ExportNamedDeclaration' &&
+      node.declaration &&
+      node.declaration.type === 'VariableDeclaration'
+    ) {
+      node.declaration.declarations.forEach(declarator => {
+        if (
+          declarator.type === 'VariableDeclarator' &&
+          declarator.id.type === 'Identifier' &&
+          declarator.init &&
+          isCreateTranslatorCall(declarator.init)
+        ) {
+          const exportName = declarator.id.name;
+          const keys = extractMessageKeys(declarator.init);
+          if (keys) {
+            translators.set(exportName, keys);
+          }
+        }
+      });
+    }
+
+    // Pattern 2: export default createTranslator(...)
+    if (
+      node.type === 'ExportDefaultDeclaration' &&
+      node.declaration &&
+      isCreateTranslatorCall(node.declaration)
+    ) {
+      const keys = extractMessageKeys(node.declaration);
+      if (keys) {
+        translators.set('default', keys);
+      }
+    }
+
+    // Pattern 3: const name = createTranslator(...) (for later re-export)
+    // We track ALL variable declarations; Pattern 1 will override if it's an export
+    if (node.type === 'VariableDeclaration') {
+      node.declarations.forEach(declarator => {
+        if (
+          declarator.type === 'VariableDeclarator' &&
+          declarator.id.type === 'Identifier' &&
+          declarator.init &&
+          isCreateTranslatorCall(declarator.init)
+        ) {
+          const name = declarator.id.name;
+          const keys = extractMessageKeys(declarator.init);
+          if (keys) {
+            // Only add to localTranslators if not already in translators
+            // (Pattern 1 would have added it to translators already)
+            if (!translators.has(name)) {
+              localTranslators.set(name, keys);
+            }
+          }
+        }
+      });
+    }
+
+    // Pattern 4: export { name } (re-export pattern)
+    if (node.type === 'ExportNamedDeclaration' && !node.declaration && node.specifiers) {
+      node.specifiers.forEach(spec => {
+        if (spec.type === 'ExportSpecifier') {
+          const localName = spec.local.name;
+          const exportedName = spec.exported.name;
+          if (localTranslators.has(localName)) {
+            translators.set(exportedName, localTranslators.get(localName));
+          }
+        }
+      });
+    }
+
```

Comment:
Although less common, there is a Pattern 5/6: an object of translators https://github.com/learningequality/kolibri/blob/9834a4dae244c4b821f094adbbf7213b825714f0/kolibri/plugins/coach/assets/src/views/common/status/statusStrings.js#L14. It will miss cases like:

```Javascript
const { randomString$ } = learnerProgressTranslators.completed;
// or
const { completed } = learnerProgressTranslators;
const { randomString$ } = completed;
```

Although this is very specific, and it may be more challenging to track. But just flagging it anyways

Reply:
I think it may be worth covering - we're already handling traversing module imports, so lookups on objects shouldn't be out of the question.